docs/operations/hook-sidecar.md

```diff
@@ -0,0 +1,145 @@
+# Hook Sidecar Container
+
+## Introduction
```

Comment:
In the introduction, you can point out what `preCloudInitIso` is used for too

Reply:
Same as https://github.com/kubevirt/user-guide/pull/751#discussion_r1446029755. I'm myself having a hard time understanding what _exactly_ it does.

Reply:
Exactly, that's why we need documentation :)
I'm sorry, i can't help you too much here. I didn't look at it too closely, only what the [cloudinit example](https://github.com/kubevirt/kubevirt/blob/main/cmd/sidecars/cloudinit/cloudinit.go) sidecar does plus testing it.